components/pedigree/resources/src/main/resources/pedigree/controller.js

```diff
@@ -1016,12 +1096,24 @@ define([
             processLinkCallback(false);
         } else {
             if (linkID == "") {
+                var deletePatientRecord = function(clearParameter) {
+                    var onDeletePatientRecord = function() {
+                        callbackOnValid(clearParameter, true);
+                    }
+                    var event = { "patientProfilesToBeRemoved": [ editor.getGraph().getPhenotipsLinkID(nodeID) ],
+                                  "callbackOnApprove": onDeletePatientRecord };
+                    document.fire("pedigree:patient:deleterequest", event);
+                }
+
                 var oldLinkID = editor.getNode(nodeID).getPhenotipsPatientId();
                 editor.getOkCancelDialogue().showWithCheckbox("<br/><b>Do you want to remove the connection between this<br/>patient </b>(" + editor.getGraph().getPatientDescription(nodeID, true) +
                         ")<b> and this pedigree node?</b><br/><br/><br/>" +
                         "<div style='margin-left: 30px; margin-right: 30px; text-align: center'>" +
                         "Please note that if you do not assign this patient to another pedigree<br/>node this patient will be removed from this family</div><br/>",
-                        'Remove the connection?', 'Clear data from this pedigree node', true, "Remove link", processLinkCallback, "Cancel", onCancelAssignPatient );
+                        'Remove the connection?', 'Clear data from this pedigree node', true,
+                        "Remove link", processLinkCallback,
+                        "Cancel", onCancelAssignPatient,
+                        "Delete patient record", deletePatientRecord);
```

Comment:
`Cancel` in the middle?

Reply:
Not sure what is the question? If you mean the porder of buttons in UI, yes, it will look like

```----------------------------------------------------
                     message text

           [Unlink] [Cencel]                 [Delete]
----------------------------------------------------```